openeo/rest/connection.py

```diff
@@ -790,25 +792,22 @@ def load_collection(
 
     imagecollection = legacy_alias(load_collection, name="imagecollection")
 
-    def create_service(self, graph: dict, type: str, **kwargs) -> dict:
+    def create_service(self, graph: dict, type: str, **kwargs) -> Service:
         # TODO: type hint for graph: is it a nested or a flat one?
         req = self._build_request_with_process_graph(process_graph=graph, type=type, **kwargs)
         response = self.post(path="/services", json=req, expected_status=201)
-        # TODO: "location" is url of the service metadata, not (base) url of service (https://github.com/Open-EO/openeo-api/issues/269)
-        # TODO: fetch this metadata and return a full metadata object instead?
-        return {
-            'url': response.headers.get('Location'),
-            'service_id': response.headers.get("OpenEO-Identifier"),
-        }
+        service_id = response.headers.get("OpenEO-Identifier")
+        return Service(service_id, self)
 
+    @deprecated("Use :py:meth:`openeo.service.Service.delete_service` instead.", version="0.7.0")
```

Comment:
next version will be at least 0.7.1, so that should be the deprecation threshold

Reply:
Fixed.